xDrip/Model/Entries/AbstractEntry.swift

```diff
@@ -12,12 +12,32 @@ import RealmSwift
 protocol AbstractEntryProtocol {
     var amount: Double { get }
     var date: Date? { get }
+    var absorptionDuration: TimeInterval? { get set }
```

Comment:
absorption duration doesn't belong to the abstract entry protocol, because it is specific for carbs and insulin. You can either add a child protocol like `AbstractAbsorbableEntry` or create a distinct protocol `Absorbable` and use both.